src/stores/prefs-store.js

```diff
@@ -0,0 +1,96 @@
+import {observable, action} from 'mobx';
+import { findByProperty } from 'util/common';
+import Torrent from './torrent';
+
+export const PrefCookieKeys = {
+  statusFilter: 'filter',
+  sortCriteria: 'sort_method',
+  sortDirection: 'sort_direction',
+  compact: 'compact_display_state'
+}
+
+export const FilterStates = [
+  {value: 0, label: 'All', persistKey: 'all'},
+  {value: 11, label: 'Active', persistKey: 'active'},
+  {value: Torrent.STATUS_DOWNLOAD, label: 'Downloading', persistKey: 'downloading'},
+  {value: Torrent.STATUS_SEED, label: 'Seeding', persistKey: 'seeding'},
+  {value: Torrent.STATUS_STOPPED, label: 'Paused', persistKey: 'paused'},
+  {value: 55, label: 'Finished', persistKey: 'finished'},
+];
+
+class PrefsStore {
+  @observable statusFilter;
+  @observable sortCriteria;
+  @observable sortDirection;
+  @observable compact;
+
+  constructor() {
+    this.rehydrate();
+  }
+
+  @action rehydrate() {
+    // status filter is saved as a string (i.e. 'all')
+    // so we need to make sure to save and read it as so.
+    const savedFilter = this.rehydrateKey(PrefCookieKeys.statusFilter, 'all');
+    this.statusFilter = findByProperty(FilterStates, 'persistKey', savedFilter).value;
+
+    this.sortCriteria = this.rehydrateKey(PrefCookieKeys.sortCriteria, 'name');
+    this.sortDirection = this.rehydrateKey(PrefCookieKeys.sortDirection, '');
+    this.compact = this.rehydrateKey(PrefCookieKeys.compact, false);
+  }
+
+  @action setStatusFilter(statusFilter) {
+    this.statusFilter = statusFilter;
+
+    // status filter is saved as a string (i.e. 'all')
+    // so we need to make sure to save and read it as so.
+    const persistAs = findByProperty(FilterStates, 'value', statusFilter).persistKey;
+    this.persistKey(PrefCookieKeys.statusFilter, persistAs);
```

Comment:
Instead of performing data serialization here, it's much preferable to move this to a [reaction](https://mobx.js.org/refguide/reaction.html).

You can check notifications example in `reactions` folder.